active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+    else
+        // Do something when an error was removed.
+
+        SaveButton.IsEnabled = Validation.GetErrors(e.Source);
+}
+```
+
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+## InputPropertyAttribute
+The `InputPropertyAttribute` helps us ensure that we only generate validation code for the property specified by this attribute. This will help reduce code bloat by not generating code for properties that will never perform validation (like width). Controls can specify multiple properties as their input property. 
+
+    /// <summary>
+    /// Name of the property that is associated with user input.
+    /// </summary>
+    attribute InputPropertyAttribute
+    {
+         string Name;
+    }
+
+## InputValidationKind  
+Determines how the control displays validation visuals.
+
+    /// <summary>
+    ///    Determines how the control displays it's validation visuals
+    /// </summary>
+    [webhosthidden]
+    enum InputValidationKind
+    {
+        /// <summary>
+        /// Let's the Control dictate how to display visuals. This is the default value.
+        /// </summary>
+        Auto,
+        /// <summary>
+        /// An error icon displays to the right of the control that displays the error messages in a tool tip.
+        /// </summary>
+        Compact,
+        /// <summary>
+        /// Text for the error messages is displayed underneath the control.  
+        /// </summary>
+        Inline,
+        /// <summary>
+        /// Validation visuals are not displayed at all.
+        /// </summary>
+        Hidden,
```

Comment:
Can you include an example of why this is relevant/useful?
I assume that it's for scenarios where the app will show errors for the view in a list together, rather than inline, but it would be good to include an explanation.

Reply:
Great questions! So originally, we had to make `Compact` the default, since having them set to `Inline` would have been a breaking change.  Specifying `Inline` means that an extra padding is added below the control so that when the errors are shown, there is no affect to layout (assuming only a single line of text).